lua/oil/util.lua

```diff
@@ -697,4 +697,54 @@ M.adapter_list_all = function(adapter, url, opts, callback)
   end)
 end
 
+M.send_to_quickfix = function(opts)
+  if type(opts) ~= "table" then
+    opts = {}
+  end
+  local oil = require("oil")
+  local dir = oil.get_current_dir()
+  if type(dir) ~= "string" then
+    return
+  end
+
+  local qf_entries = {}
+  for i = 1, vim.fn.line("$") do
+    local entry = oil.get_entry_on_line(0, i)
+    if
+      entry
+      and (
+        entry.type == "file" and opts.files ~= false
+        or entry.type == "directory" and opts.directories ~= false
+      )
+    then
+      local qf_entry = {
+        filename = dir .. entry.name,
+        lnum = 1,
+        col = 1,
+        text = entry.name,
+      }
+      table.insert(qf_entries, qf_entry)
+    end
+  end
```

Comment:
This looks like it adds every entry in the buffer to the quickfix. Maybe we should detect if we're in visual mode and only add the selected entries?

Reply:
It only sends selected items when in visual mode now. I reused the logic you implemented in `lua/oil/init.lua:459` and gathered it in  `get_visual_range` function in util module.